### DISCUSSION TITLE
Wallet connect v2 metamask signature request doesn't appear on metamask mobile.

### DISCUSSION THREAD

**Discussion by ghost** [WalletConnect/walletconnect-monorepo#3215] [Developer Support]
Packages version :
Viem: 1.3.0
Wagmi: 1.3.9
Metamask mobile: 7.3.0 on IOS

Using walletConnect v2 as a connector, I can connect to a MetaMask mobile wallet using the QR code then when I use the signMessage() action nothing happens, it doesn't trigger an error also (at least the try and catch doesn't catch anything).
It works with Rainbow wallet, trust wallet etc.

**Comment by ghost** [comment]
Metamask 7.2.0 sorry

**Comment by ghost** [comment]
Adding some context here.
Other users are affected, but we can't pinpoint where the source issue is from.
https://github.com/MetaMask/metamask-mobile/issues/6835

**Comment by Haavi97** [comment]
I have a similar error which might be related. 

In my application first I connect using web3-react implementation Wallet Connect V2 scanning the QR code and it works without problem, but then as a second step the app needs to get the signature from the user. 
For that I get the signer using the web3-react provider that is already connected through wallet connect.
Then, after getting the signer is when I ask for the signature:
```
const signer = provider.getSigner();
signature = await signer.signMessage(msg);
```
But that throws an error
{context: 'client'} 'Invalid Id'
Here a screenshot:
<img width="947" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/57443449/b6195b0e-e486-4769-95ec-41c4a5e96a62">



But I realized that if you try it a second time it actually works, so I did a dirty work around until I find what is causing that error.
```
const signer = provider.getSigner();
try {
  signature = await signer.signMessage(msg);
} catch (e) {
  console.log(e);
  console.log('try again');
  signature = await signer.signMessage(msg);
}
```

Any idea why this might be happening?

**Comment by glitch-txs** [comment]
Hi, which method are you using for signing?

**Reply by Haavi97** [reply]
What do you mean?

I am just using `signMessage` and passing a string with the text to sign.

**Reply by glitch-txs** [reply]
where is this function coming from? there are different types of signature.

**Reply by Haavi97** [reply]
It's from the provider that comes from using the hook `useWeb3React` which is tied to the Wallet Connect connctor from the web3-react implementation. 

As for the specific method, I don't really know. The web3-react library documentation is practically useless.

I have been trying to navigate what is actually behing but I am not sure. 
If you mean `personal_sign `or `eth_sign `I think it doesn't really matter since for both are using the same `signMessage ` method as far as I know. 
It my from here: https://docs.ethers.org/v5/api/signer/#Signer-signMessage 
??

**Reply by glitch-txs** [reply]
this actually seems unrelated to MetaMask, you can test here https://react-app.walletconnect.com/
 could you open an issue/discussion on web3-react repo and ping me there to follow up?
Also do you have any public repo we can check? Thanks

**Reply by glitch-txs** [reply]
I see it takes some config, are you passing "eth_sign" in the method option when initializing the construtor? I'm not very familiar with web3-react atm tbh. But you should add eth_sign as supported methods in the WalletConnect configuration.
